tensorflow/python/lib/io/file_io.i

```diff
@@ -237,6 +237,67 @@ string ReadFromStream(tensorflow::io::BufferedInputStream* stream,
   return result;
 }
 
+string ReadFromStreamUTF8(tensorflow::io::BufferedInputStream* stream,
+                          size_t bytes,
+                          TF_Status* status) {
+  tensorflow::tstring result;
+  size_t total = 0;
+  while (total < bytes) {
+    // Avoid over read:
+    // 1) read at least bytes (utf8 could be 1/2/3/4)
+    // 2) if string ends with partial utf8 (2/3/4), read remaining (< 4) bytes.
+    // 3) repeat until bytes == utf8 chars
+
+    // 1) read at least bytes - total
+    size_t bytes_to_read = bytes - total;
+    tensorflow::tstring result_read;
+    tensorflow::Status s = stream->ReadNBytes(bytes_to_read, &result_read);
+    if (!s.ok() && s.code() != tensorflow::error::OUT_OF_RANGE) {
+      Set_TF_Status_from_Status(status, s);
+      result.clear();
+      return result;
+    }
+    result += result_read;
+    if (s.code() == tensorflow::error::OUT_OF_RANGE) {
+      break;
+    }
+    // 2) find partial utf8, and read remain (< 4) bytes.
+    size_t remain = 0;
+    for (size_t i = 0; i < result_read.size(); i++) {
+      if (remain > 0) {
+        remain--;
+        continue;
+      }
+      if ((result_read[i] & 0xe0) == 0xc0) {
```

Comment:
Let's extract the `0xe0`, `0xc0`, `0xf0` and `0xf8` magic constants to same named constants to increase readability